src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/FermiLiteAssemblyHandler.java

```diff
@@ -24,51 +26,509 @@
+        final FermiLiteAssembly initialAssembly = assembler.createAssembly(readsList);
         final int secondsInAssembly = (int)((System.currentTimeMillis() - timeStart + 500)/1000);
+        if ( initialAssembly.getNContigs() == 0 ) {
+            return new AlignedAssemblyOrExcuse(intervalID, "no assembly -- no contigs produced by assembler.");
+        }
+
+        // patch up the assembly to improve contiguity
+        final FermiLiteAssembly assembly = reviseAssembly(initialAssembly, removeShadowedContigs, expandAssemblyGraph);
+
+        // record the assembly as a GFA, if requested
         if ( fastqDir != null && writeGFAs ) {
-            final String gfaName =  String.format("%s/%s.gfa",fastqDir, AlignedAssemblyOrExcuse.formatAssemblyID(intervalID));
+            final String gfaName =  String.format("%s/%s.gfa", fastqDir, assemblyName);
             try ( final OutputStream os = BucketUtils.createFile(gfaName) ) {
                 assembly.writeGFA(os);
             }
             catch ( final IOException ioe ) {
                 throw new GATKException("Can't write "+gfaName, ioe);
             }
         }
-        final List<byte[]> tigSeqs =
-                assembly.getContigs().stream()
-                        .map(FermiLiteAssembly.Contig::getSequence)
-                        .collect(SVUtils.arrayListCollector(assembly.getNContigs()));
+
+        // align the assembled contigs to the genomic reference
         try ( final BwaMemAligner aligner = new BwaMemAligner(BwaMemIndexCache.getInstance(alignerIndexFile)) ) {
             aligner.setIntraCtgOptions();
-            final List<List<BwaMemAlignment>> alignments = aligner.alignSeqs(tigSeqs);
+            final List<byte[]> sequences =
+                    assembly.getContigs().stream()
+                            .map(Contig::getSequence)
+                            .collect(SVUtils.arrayListCollector(assembly.getNContigs()));
+            final List<List<BwaMemAlignment>> alignments = aligner.alignSeqs(sequences);
             return new AlignedAssemblyOrExcuse(intervalID, assembly, secondsInAssembly, alignments);
         }
     }
+
+    @VisibleForTesting
+    static FermiLiteAssembly reviseAssembly( final FermiLiteAssembly initialAssembly,
+                                             final boolean removeShadowedContigs,
+                                             final boolean expandAssemblyGraph ) {
+        final FermiLiteAssembly unshadowedAssembly =
+                removeShadowedContigs ? removeShadowedContigs(initialAssembly) : initialAssembly;
+        return expandAssemblyGraph ? expandAssemblyGraph(removeUnbranchedConnections(unshadowedAssembly)) : unshadowedAssembly;
+    }
+
+    @VisibleForTesting
+    static FermiLiteAssembly removeShadowedContigs( final FermiLiteAssembly assembly ) {
+        final int kmerSize = 31;
+        final double maxMismatchRate = .05;
```

Comment:
Yeah, I wasn't thinking that we'd be able to assemble them perfectly, but was worried that doing this might introduce misassemblies in some cases. Just speculative, though.

Reply:
I tested with a 1% max mismatch rate, and the results don't differ materially, and are arguably a tiny bit better.  So I'll make that change.  Let's have a brief chalk talk about seg dups when we're all in the office.